### PR TITLE
docs note: `azurerm_cognitive_account`, not all `kind` support `storage` block

### PR DESCRIPTION
**Pull request by zioproto**
docs note: `azurerm_cognitive_account`, not all `kind` support `storage` block

This needs more visibility: a lot of folks are testing OpenAI on Azure, and it is confusing what to do about the storage block.

### CODE REVIEW

File: website/docs/r/cognitive_account.html.markdown [hunk 1]
```diff
@@ -138,6 +138,8 @@ A `storage` block supports the following:
 
 * `identity_client_id` - (Optional) The client ID of the managed identity associated with the storage resource.
 
+~> **NOTE:** Not all `kind` support a `storage` block. For example the `kind` `OpenAI` does not support it.
```
Comment (stephybun) [inline website/docs/r/cognitive_account.html.markdown:141]: @zioproto is there some MSFT documentation that we can link here with a definitive list of what cognitive account kinds support storage? If not then I think we should rephrase this to be more specific
```suggestion
~> **NOTE:** Cognitive Accounts with `kind` `OpenAI` do not support a `storage` block.
```

**Review by stephybun — COMMENTED** [review]
Thanks for this @zioproto. I left a comment in-line if you could take a look please.

File: website/docs/r/cognitive_account.html.markdown (hunk 1 above)
Comment (zioproto) [reply website/docs/r/cognitive_account.html.markdown:141]: @felixwa Are you able to help me with this ask ? Is it just `OpenAI` that does not support this ? Thanks

Comment (zioproto) [reply website/docs/r/cognitive_account.html.markdown:141]: @stephybun it seems there is not a MSFT documentation page to link that publishes this definitive list. Should we just keep the generic sentence ?
```
Not all `kind` support a `storage` block.
```

I added the OpenAI example because it is very popular and a lot of customers ask about what the storage block is for.

Please advice what changes are necessary to get this merged. Thanks

**Review by stephybun — APPROVED** [review]
Thanks @zioproto LGTM 👍
